nano/lib/utility.cpp

```diff
@@ -238,9 +238,9 @@ thread ([this]() {
 
 void nano::worker::run ()
 {
+	std::unique_lock<std::mutex> lk (mutex);
```

Comment:
Not sure how I feel about this one, because it's now going to be protecting `stopped` which it doesn't need to. If you are going to do this then can you make `stopped` non-atomic and protect it in `stop`. It's cheap to make a `unique_lock` fyi, it's just a pointer to a mutex and a `bool`. The `lk.lock ()` in the loop is superfluous so probably shouldn't have been there to begin with.

Reply:
Thanks, did not know about this. Did your other comment disappear somehow? I applied both

Reply:
Yeh, had a talk with @cryptocode and didn't want to force you to do it :P